pkg/cache/cluster.go

```diff
@@ -1240,16 +1296,65 @@ func (c *clusterCache) processEvent(event watch.EventType, un *unstructured.Unst
 		return
 	}
 
+	c.eventMetaCh <- eventMeta{event, un}
+}
+
+func (c *clusterCache) processEvents() {
+	log := c.log.WithValues("fn", "processItems")
+	log.V(1).Info("Start processing events")
+
 	c.lock.Lock()
-	defer c.lock.Unlock()
-	existingNode, exists := c.resources[key]
-	if event == watch.Deleted {
-		if exists {
-			c.onNodeRemoved(key)
+	ch := c.eventMetaCh
+	c.lock.Unlock()
+
+	eventMetas := make([]eventMeta, 0)
+	ticker := time.NewTicker(c.eventProcessingInterval)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case em, ok := <-ch:
+			if !ok {
+				log.V(1).Info("Event processing channel closed, finish processing")
+				return
+			}
+			eventMetas = append(eventMetas, em)
+		case <-ticker.C:
+			if len(eventMetas) > 0 {
+				c.processEventsBatch(eventMetas)
+				eventMetas = eventMetas[:0]
```

Comment:
Why assign `eventMetas[:0]` instead of nil or empty slice?

Reply:
Thank you for review @andrii-korotkov-verkada 

Setting `eventMetas` to `eventMetas[:0]` retains the original slice’s underlying array, which can help reduce memory allocations if the slice is reused frequently. Assigning `nil` would release the underlying array, and creating a new empty slice would lead to additional allocations. This approach allows us to efficiently clear the slice while keeping the capacity intact for future use.